tools/src/main/java/org/apache/kafka/tools/ProducerPerformance.java

```diff
@@ -486,4 +536,33 @@ public void onCompletion(RecordMetadata metadata, Exception exception) {
         }
     }
 
+    public static class CreateUnbalancePartitioner implements Partitioner {
+        private List<Integer> partitionsInBrokers;
+
+        @Override
+        public void configure(Map<String, ?> configs) {
+            String partitionsInBrokersStr = (String) configs.get("partitionsInBrokers");
+            if (partitionsInBrokersStr == null || partitionsInBrokersStr.isEmpty()) {
+                throw new IllegalArgumentException("The list of partitions in brokers cannot be null or empty.");
+            }
+
+            this.partitionsInBrokers = Arrays.stream(partitionsInBrokersStr.split(","))
+                .map(Integer::parseInt)
+                .collect(Collectors.toList());
+        }
+
+        @Override
+        public int partition(String topic, Object key, byte[] keyBytes, Object value, byte[] valueBytes, Cluster cluster) {
```

Comment:
You seem to have enough information from the `cluster` parameter to determine the unique node ID to send all the data.

So, consider moving all the partition selection logic to this partitioner, you even don't need to set any key for records.